src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+        max_size.append(max_size_i)
+    max_size = tuple(max_size)
+
+
+# copied from original implementation
+# TODO this has to go away!
+def _max_by_axis(the_list):
+    # type: (List[List[int]]) -> List[int]
+    maxes = the_list[0]
+    for sublist in the_list[1:]:
+        for index, item in enumerate(sublist):
+            maxes[index] = max(maxes[index], item)
+    return maxes
+
+
+# copied from original implementation
+# TODO this has to go away!
+# This function pads images to match the max one in the list (similar to padding sequences)
+# however it is not clear to me why this is needed in the first place. The only case in which we need this
+# is when a batch of targets has a different masks-size. This case should be handled in the `collate_fn`
+# not inside the model. In the implementation I've assumed targets in a batch shares the same spatial size
+# for both images and masks
+def nested_tensor_from_tensor_list(tensor_list: List[Tensor]):
+    # TODO (authors) make this more general
+    if tensor_list[0].ndim == 3:
+        if torchvision._is_tracing():
+            # nested_tensor_from_tensor_list() does not export well to ONNX
+            # call _onnx_nested_tensor_from_tensor_list() instead
+            return _onnx_nested_tensor_from_tensor_list(tensor_list)
+
+        # TODO (authors) make it support different-sized images
+        max_size = _max_by_axis([list(img.shape) for img in tensor_list])
+        # min_size = tuple(min(s) for s in zip(*[img.shape for img in tensor_list]))
+        batch_shape = [len(tensor_list)] + max_size
+        b, c, h, w = batch_shape
+        dtype = tensor_list[0].dtype
+        device = tensor_list[0].device
+        tensor = torch.zeros(batch_shape, dtype=dtype, device=device)
+        mask = torch.ones((b, h, w), dtype=torch.bool, device=device)
+        for img, pad_img, m in zip(tensor_list, tensor, mask):
+            pad_img[: img.shape[0], : img.shape[1], : img.shape[2]].copy_(img)
+            m[: img.shape[1], : img.shape[2]] = False
+    else:
+        raise ValueError("not supported")
+    return NestedTensor(tensor, mask)
+
+
+# copied from original implementation
+def is_dist_avail_and_initialized():
+    if not dist.is_available():
+        return False
+    if not dist.is_initialized():
+        return False
+    return True
+
+
+# refactored from original implementation
+def dice_loss(inputs: Tensor, targets: Tensor, num_masks: float) -> Tensor:
+    r"""
+    Compute the DICE loss, similar to generalized IOU for masks as follow
+
+    .. math::
+        \mathcal{L}_{\text{dice}(x, y) = 1 - \frac{2 * x \cap y }{x \cup y + 1}}
+
+    In practice, since `targets` is a binary mask, (only 0s and 1s), dice can be computed as follow
+
+    .. math::
+        \mathcal{L}_{\text{dice}(x, y) = 1 - \frac{2 * x * y }{x + y + 1}}
+
+    Args:
+        inputs (Tensor): A tensor representing a mask
```

Comment:
```suggestion
        inputs (`torch.Tensor`): A tensor representing a mask
```
Putting it here once, but is applicable to every types in docstrings.